cg/meta/workflow/nf_analysis.py

```diff
@@ -315,18 +316,39 @@ def get_deliverables_template_content() -> list[dict]:
     def get_deliverables_for_case(self, case_id: str) -> PipelineDeliverables:
         """Return PipelineDeliverables for a given case."""
         deliverable_template: list[dict] = self.get_deliverables_template_content()
-        sample_id: str = self.status_db.get_samples_by_case_id(case_id).pop().internal_id
+        samples: list[Sample] = self.status_db.get_samples_by_case_id(case_id=case_id)
+        case_path: str = str(self.get_case_path(case_id=case_id))
         files: list[FileDeliverable] = []
-        for file in deliverable_template:
-            for deliverable_field, deliverable_value in file.items():
-                if deliverable_value is None:
-                    continue
-                file[deliverable_field] = file[deliverable_field].replace("CASEID", case_id)
-                file[deliverable_field] = file[deliverable_field].replace("SAMPLEID", sample_id)
-                file[deliverable_field] = file[deliverable_field].replace(
-                    "PATHTOCASE", str(self.get_case_path(case_id=case_id))
+
+        for sample in samples:
+            sample_id: str = sample.internal_id
+            sample_name: str = sample.name
+
+            for file in deliverable_template:
+                deliverables = dict(file)
+                for deliverable_field, deliverable_value in file.items():
+                    if deliverable_value is None:
+                        continue
+
+                    deliverables[deliverable_field] = deliverables[deliverable_field].replace(
+                        "CASEID", case_id
+                    )
+                    deliverables[deliverable_field] = deliverables[deliverable_field].replace(
+                        "SAMPLEID", sample_id
+                    )
+                    deliverables[deliverable_field] = deliverables[deliverable_field].replace(
+                        "SAMPLENAME", str(sample_name)
+                    )
+                    deliverables[deliverable_field] = deliverables[deliverable_field].replace(
+                        "PATHTOCASE", case_path
```

Comment:
I agree. This function could look like this:
```python
def get_deliverables_for_case(self, case_id: str) -> PipelineDeliverables:
        """Return PipelineDeliverables for a given case."""
        deliverable_template: list[dict] = self.get_deliverables_template_content()
        samples: list[Sample] = self.status_db.get_samples_by_case_id(case_id=case_id)

        for sample in samples:
            files: list[FileDeliverable] = self.get_deliverables_for_sample(sample=sample, case_id=case_id, template=deliverable_template)

        return PipelineDeliverables(files=files)
```
and `get_deliverables_for_sample`:
```python
def get_deliverables_for_sample(sample: Sample, case_id: str, template: list[dict]) -> list[FileDeliverable]:
    """Some docstring."""
    sample_id: str = sample.internal_id
    sample_name: str = sample.name
    case_path: str = str(self.get_case_path(case_id=case_id))
    for file in template:
        deliverables = file.copy()
        for deliverable_field, deliverable_value in file.items():
            if deliverable_value and deliverables not in files
                deliverables[deliverable_field] = deliverables[deliverable_field].replace("CASEID", case_id).replace(...).replace(...).replace(...)
                files.append(FileDeliverable(**deliverables))
    return files
```

Reply:
But for another PR the templates fro RNAFUSION and taxprofiler should be modified so that they can be formatted easily with `.format` (Using the curly brackets for variables to replace {})

Reply:
I updated the PR with your suggestions.